tests/general/mmap.c

```diff
@@ -0,0 +1,87 @@
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <limits.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#ifdef __GLIBC__
+#include <sys/user.h>
+#endif
+#include <unistd.h>
+
+#define perror_and_exit(message) \
+  do {                           \
+    perror(message);             \
+    return EXIT_FAILURE;         \
+  } while (0)
+
+#define OFFSET 10726
+#define LENGTH 143
+
+int main(int argc, char *argv[]) {
+  if (argc != 2) {
+    fprintf(stderr, "usage: %s <dir>\n", argv[0]);
+    return EXIT_FAILURE;
+  }
+  char *filename;
+  if (asprintf(&filename, "%s/input.txt", argv[1]) == -1) {
+    fprintf(stderr, "can't allocate filename");
+    return EXIT_FAILURE;
+  }
+
+  int fd = open(filename, O_RDONLY);
+  if (fd < 0)
+    perror_and_exit("open");
+
+  struct stat stat_buf;
+  if (fstat(fd, &stat_buf) != 0)
+    perror_and_exit("fstat");
+
+  off_t offset = OFFSET;
+  if (offset < 0) {
+    fprintf(stderr, "negative offset\n");
+    return EXIT_FAILURE;
+  }
+  if (offset > (off_t)SIZE_MAX) {
+    fprintf(stderr, "offset overflow\n");
+    return EXIT_FAILURE;
+  }
```

Comment:
Why not drop these checks too, since the value is constant?    lgtm either way

Reply:
The offset value is effectively a parameter to the test, so it's nice to sanity-check it.